salt/utils/gitfs.py

```diff
@@ -1740,7 +1744,9 @@ def _fetch(self):
         else:
             log.debug('%s remote \'%s\' is up-to-date', self.role, self.id)
         refs_post = self.repo.listall_references()
-        cleaned = self.clean_stale_refs(local_refs=refs_post)
+        cleaned = []
+        if not HAS_PYGIT2_FETCH_PRUNE:
```

Comment:
Instead of this defining `cleaned` twice, who about reverting this particular change and doing this?

```diff
diff --git a/salt/utils/gitfs.py b/salt/utils/gitfs.py
index 286657c72c..9e499b52a3 100644
--- a/salt/utils/gitfs.py
+++ b/salt/utils/gitfs.py
@@ -1583,11 +1583,19 @@ class Pygit2(GitProvider):
         '''
         Clean stale local refs so they don't appear as fileserver environments
         '''
+        try:
+            if pygit2.GIT_FETCH_PRUNE:
+                # Don't need to clean anything, pygit2 can do it by itself
+                return []
+        except AttributeError:
+            # However, only in 0.26.2 and newer
+            pass
         if self.credentials is not None:
             log.debug(
-                'pygit2 does not support detecting stale refs for '
-                'authenticated remotes, saltenvs will not reflect '
-                'branches/tags removed from remote \'%s\'', self.id
+                'The installed version of pygit2 (%s) does not support '
+                'detecting stale refs for authenticated remotes, saltenvs '
+                'will not reflect branches/tags removed from remote \'%s\'',
+                PYGIT2_VERSION, self.id
             )
             return []
         return super(Pygit2, self).clean_stale_refs()
```